packages/cadl-ranch-specs/http/type/model/inheritance/multiple-discriminator/main.tsp

```diff
@@ -0,0 +1,222 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+
+using TypeSpec.Http;
+
+@doc("Illustrates multiple level inheritance with multiple discriminators.")
+@scenarioService("/type/model/inheritance/multiple-discriminator")
+namespace Type.Model.Inheritance.MultipleDiscriminator;
+
+@doc("This is base model for polymorphic multiple levels inheritance with a discriminator.")
+@discriminator("kind")
+model Fish {
+  age: int32;
+}
+
+@doc("The second level model in polymorphic multiple levels inheritance and it defines a new discriminator.")
+@discriminator("sharktype")
+model Shark extends Fish {
+  kind: "shark";
+}
+
+@doc("The second level model in polymorphic multiple levels inheritance which contains references to other polymorphic instances.")
+model Salmon extends Fish {
+  kind: "salmon";
+  friends?: Fish[];
+  hate?: Record<Fish>;
+  partner?: Fish;
+}
+
+@doc("The third level model SawShark in polymorphic multiple levels inheritance.")
+model SawShark extends Shark {
+  sharktype: "saw";
+}
+
+@doc("The third level model GoblinShark in polymorphic multiple levels inheritance.")
+model GoblinShark extends Shark {
+  sharktype: "goblin";
+}
+
+@scenario
+@route("/model")
+@scenarioDoc("""
+Generate and receive polymorphic model in multiple levels inheritance with 2 discriminators.
+Expected response body:
+```json
+{"age": 1, "kind": "shark",  "sharktype": "goblin"}
+```
+""")
+@get
+op getModel(): Fish;
+
+@scenario
+@route("/model")
+@scenarioDoc("""
+Generate and send polymorphic model in multiple levels inheritance with 2 discriminators.
+Expected input body:
+```json
+{"age": 1, "kind": "shark",  "sharktype": "goblin"}
+```
+""")
+@put
+op putModel(@body input: Fish): OkResponse;
```

Comment:
```suggestion
op putModel(@body input: Fish): NoContentResponse;
```

return NoContentresponse or `void` for operations without body.

Reply:
fixed